pkg/providers/instance/instance.go

```diff
@@ -87,6 +89,8 @@ type Provider interface {
 	// CreateTags(context.Context, string, map[string]string) error
 	Update(context.Context, string, armcompute.VirtualMachineUpdate) error
 	GetNic(context.Context, string, string) (*armnetwork.Interface, error)
+	DeleteNic(context.Context, string) error
+	ListNics(context.Context) ([]*armnetwork.Interface, error)
```

Comment:
Future refactor here would be to factor out these calls into a network interface provider.